app/lib/pages/settings/notification.dart

```diff
@@ -205,6 +231,65 @@ class _HomeworkNotificationsTimeTile extends StatelessWidget {
   }
 }
 
+class _HomeworkNotificationsTimeTileTrailing extends StatelessWidget {
+  const _HomeworkNotificationsTimeTileTrailing();
+
+  @override
+  Widget build(BuildContext context) {
+    final isUnlocked = context
+        .read<SubscriptionService>()
+        .hasFeatureUnlocked(SharezonePlusFeature.changeHomeworkReminderTime);
+    return isUnlocked ? const SizedBox() : const SharezonePlusChip();
+  }
+}
+
+class _ChangeHomeworkNotificationsTimePlusAd extends StatelessWidget {
+  const _ChangeHomeworkNotificationsTimePlusAd();
+
+  @override
+  Widget build(BuildContext context) {
+    final buttonShape = RoundedRectangleBorder(
+      borderRadius: BorderRadius.circular(12.5),
+    );
+    return AlertDialog(
```

Comment:
I'll later make a more abstract widget in the `sharezone_widgets` package to be able to reuse this dialog for other features. Later on, we can also implement in this dialog the logic for displaying a short demo video (mobile video for mobile devices and landscape video for landscape devices).

Reply:
https://github.com/SharezoneApp/sharezone-app/pull/1008